tests/jax/components/training/step_test.py

```diff
@@ -0,0 +1,292 @@
+    trainer_step = DefaultTrainerStep()
+    old_timestamp = mock_trainer.store.timestamp
+    trainer_step.on_training_step(trainer=mock_trainer)
+
+    assert mock_trainer.store.timestamp > old_timestamp
+
+    assert list(mock_trainer.store.trainer_parameter_client.params.keys()) == [
+        "trainer_steps",
+        "trainer_walltime",
+    ]
+    assert mock_trainer.store.trainer_parameter_client.params["trainer_steps"] == 1
+
+    assert (
+        int(mock_trainer.store.trainer_parameter_client.params["trainer_walltime"]) > 0
+    )
+
+    assert mock_trainer.store.trainer_parameter_client.call_set_and_get_async is True
+
+    assert mock_trainer.store.trainer_logger.written == {"next_sample": 2, "sample": 1}
+
+
+def test_on_training_step_without_timestamp(mock_trainer: Trainer) -> None:
+    """Test on_training_step method from TrainerStep case of no timestamp"""
+    trainer_step = DefaultTrainerStep()
+    del mock_trainer.store.timestamp
+    trainer_step.on_training_step(trainer=mock_trainer)
+
+    assert mock_trainer.store.timestamp != 0
+
+    assert list(mock_trainer.store.trainer_parameter_client.params.keys()) == [
+        "trainer_steps",
+        "trainer_walltime",
+    ]
+    assert mock_trainer.store.trainer_parameter_client.params["trainer_steps"] == 1
+    assert (
+        int(mock_trainer.store.trainer_parameter_client.params["trainer_walltime"]) == 0
+    )
+
+    assert mock_trainer.store.trainer_parameter_client.call_set_and_get_async is True
+
+    assert mock_trainer.store.trainer_logger.written == {"next_sample": 2, "sample": 1}
+
+
+def test_mapg_with_trust_region_step_initiator() -> None:
+    """Test constructor of MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    assert mapg_with_trust_region_step.config.discount == 0.99
+
+
+def test_on_training_init_start(mock_trainer: MockTrainer) -> None:
+    """Test on_training_init_start method from MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    mapg_with_trust_region_step.on_training_init_start(trainer=mock_trainer)
+
+    assert mock_trainer.store.full_batch_size == 4
+
+
+def test_on_training_step_fn(mock_trainer: MockTrainer) -> None:
+    """Test on_training_step_fn method from MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    del mock_trainer.store.step_fn
+    mapg_with_trust_region_step.on_training_step_fn(trainer=mock_trainer)
+
+    assert callable(mock_trainer.store.step_fn)
+
+
+def test_step(mock_trainer: MockTrainer, dummy_sample: DummySample) -> None:
+    """Test step function"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    del mock_trainer.store.step_fn
+    mapg_with_trust_region_step.on_training_step_fn(trainer=mock_trainer)
+    old_key = mock_trainer.store.key
+    with jax.disable_jit():
```

Comment:
Just out of curiosity. Why is JIT disabled here?

Reply:
because before step function we have @jit and this made the dummy sample we have made which is the input of the step function not being accepted and I had this error:

Reply:
Ah, I see. Should we not rather try and fix the dummy sample then? As the real system uses jit and therefore we are slightly misaligned with our test here.

Reply:
Or do we test the jitted step function at some later stage?

Reply:
I tried that before and I couldn't find any solution 😓 I will try to look for further solutions!

Reply:
nope we didn't however, good updates I find a way to run samples without jit thanks to the data sample that @RuanJohn send it previously! Still have small issue I will try to fix it and commit

Reply:
Done! Thanks for the help

Reply:
Great, thanks @OmaymaMahjoub !